lib/caching-precompiler.js

```diff
@@ -30,15 +31,26 @@ CachingPrecompiler.prototype._factory = function (cacheDir) {
 		// Extract existing source maps from the code.
 		var sourceMap = convertSourceMap.fromSource(code) || convertSourceMap.fromMapFileSource(code, path.dirname(filename));
 
-		return {
-			presets: [presetStage2, presetES2015],
-			plugins: [powerAssert, transformRuntime],
+		var baseOptions = {
 			filename: filename,
 			sourceMaps: true,
 			ast: false,
-			babelrc: false,
-			inputSourceMap: sourceMap && sourceMap.toObject()
+			inputSourceMap: sourceMap && sourceMap.toObject(),
+			babelrc: false
 		};
+
+		if (!babelConfig || babelConfig === 'default') {
+			return objectAssign({}, baseOptions, {
+				presets: [presetStage2, presetES2015],
+				plugins: [powerAssert, transformRuntime]
+			});
+		}
+
+		if (babelConfig === 'inherit') {
+			return objectAssign({}, baseOptions, {babelrc: true});
```

Comment:
As discussed, we will want to ensure `powerAssert` is added as a plugin regardless.